src/librustc_mir/interpret/validity.rs

```diff
@@ -309,11 +309,14 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                         CheckInAllocMsg::InboundsTest,
                     ),
                     self.path,
-                    err_ub!(PointerOutOfBounds { .. }) |
-                    err_ub!(AlignmentCheckFailed { .. }) |
                     err_ub!(DanglingIntPointer(..)) |
+                    err_ub!(PointerUseAfterFree(..)) |
                     err_unsup!(ReadBytesAsPointer) =>
-                        { "dangling or unaligned vtable pointer in wide pointer or too small vtable" },
+                        { "dangling vtable pointer in wide pointer" },
+                    err_ub!(AlignmentCheckFailed { .. }) =>
+                        { "unaligned vtable pointer in wide pointer" },
+                    err_ub!(PointerOutOfBounds { .. }) =>
+                        { "too small vtable" },
```

Comment:
With the new macro, giving more precise errors is so easy that I just did it here. :D